test/e2e/virtual/apiexport/authorizer_test.go

```diff
@@ -0,0 +1,339 @@
+	require.NoError(t, err)
+	user3KcpClient, err := kcpclientset.NewForConfig(framework.UserConfig("user-3", rest.CopyConfig(cfg)))
+	require.NoError(t, err)
+	user2DynamicClusterClient, err := kcpdynamic.NewClusterDynamicClientForConfig(framework.UserConfig("user-2", rest.CopyConfig(cfg)))
+	require.NoError(t, err)
+	user3DynamicClusterClient, err := kcpdynamic.NewClusterDynamicClientForConfig(framework.UserConfig("user-3", rest.CopyConfig(cfg)))
+	require.NoError(t, err)
+	kubeClient, err := kubernetes.NewForConfig(rest.CopyConfig(cfg))
+	require.NoError(t, err)
+
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClient, org, []string{"user-1", "user-2", "user-3"}, nil, []string{"access"})
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClient, serviceProvider1Workspace, []string{"user-1"}, nil, []string{"admin", "access"})
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClient, serviceProvider2Workspace, []string{"user-2"}, nil, []string{"admin", "access"})
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClient, tenantWorkspace, []string{"user-3"}, nil, []string{"admin", "access"})
+	framework.AdmitWorkspaceAccess(t, ctx, kubeClient, tenantShadowCRDWorkspace, []string{"user-3"}, nil, []string{"admin", "access"})
+
+	apifixtures.CreateSheriffsSchemaAndExport(ctx, t, serviceProvider1Workspace, user1KcpClient, "wild.wild.west", "")
+
+	t.Logf("Setting maximal permission policy on APIExport %s|%s", serviceProvider1Workspace, "wild.wild.west")
+	export, err := user1KcpClient.ApisV1alpha1().APIExports().Get(logicalcluster.WithCluster(ctx, serviceProvider1Workspace), "wild.wild.west", metav1.GetOptions{})
+	require.NoError(t, err, "error getting APIExport %s|%s", serviceProvider1Workspace, export.Name)
+
+	patchedExport := export.DeepCopy()
+	patchedExport.Spec.MaximalPermissionPolicy = &apisv1alpha1.MaximalPermissionPolicy{Local: &apisv1alpha1.LocalAPIExportPolicy{}}
+	mergePatch, err := jsonpatch.CreateMergePatch(encodeJSON(t, export), encodeJSON(t, patchedExport))
+	require.NoError(t, err)
+	_, err = user1KcpClient.ApisV1alpha1().APIExports().Patch(logicalcluster.WithCluster(ctx, serviceProvider1Workspace), export.Name, types.MergePatchType, mergePatch, metav1.PatchOptions{})
+	require.NoError(t, err, "error patching APIExport %s|%s", serviceProvider1Workspace, export.Name)
+
+	t.Logf("get the sheriffs apiexport's generated identity hash")
+	identityHash := ""
+	framework.Eventually(t, func() (done bool, str string) {
+		sheriffExport, err := user1KcpClient.ApisV1alpha1().APIExports().Get(logicalcluster.WithCluster(ctx, serviceProvider1Workspace), "wild.wild.west", metav1.GetOptions{})
+		if err != nil {
+			return false, fmt.Sprintf("error while waiting to get API export: %v", err)
+		}
+		if conditions.IsTrue(sheriffExport, apisv1alpha1.APIExportIdentityValid) {
+			identityHash = sheriffExport.Status.IdentityHash
+			return true, ""
+		}
+		return false, "waiting for API export identity to be valid"
```

Comment:
It can be nice to return the reason & message on the condition here so it's clear from logs what went wrong

Reply:
indeed, changed to
```
return false, fmt.Sprintf("waiting for API export identity to be valid: %+v", conditions.Get(sheriffExport, apisv1alpha1.APIExportIdentityValid))
```